js/oai.ts

```diff
@@ -31,17 +35,29 @@ export interface OpenAIAuth {
   openAiBaseUrl?: string;
   openAiDefaultHeaders?: Record<string, string>;
   openAiDangerouslyAllowBrowser?: boolean;
+  /**
+    If present, use [Azure OpenAI Service](https://learn.microsoft.com/en-us/azure/ai-services/openai/)
+    instead of OpenAI.
+   */
+  azureOpenAi?: AzureOpenAiAuth;
+}
+
+export interface AzureOpenAiAuth {
+  apiKey: string;
+  endpoint: string;
+  apiVersion: string;
 }
 
 export function extractOpenAIArgs<T extends Record<string, unknown>>(
-  args: OpenAIAuth & T,
+  args: OpenAIAuth & T
```

Comment:
The pre-commit hooks should fix this (enforce that a comma is present). If you run `make develop` in the repo, it'll install them.

Reply:
```suggestion
  args: OpenAIAuth & T,
```

note: i had to do this manually. precommit hook didn't catch